### PR TITLE
Fix: USA Spy Drone can no longer be selected, but attacked, attempt 2

### LINKED ISSUE #115: USA Spy Drone (General Power) is selectable and exploitable
Investigate whether or not USA Spy Drone needs to be selectable at all

### PR DESCRIPTION
**Pull request by commy2**
* Fixes #115
* Relates to #1338

This change makes the USA Spy Drone no longer selectable, but still attackable. It behaves like vehicle drones.

### CODE REVIEW

**Comment by commy2** [comment]
Hovering mouse over it will still show health.

It will still be automatically attacked by idle units if detected.

You can still left click on it and units will move in range to engage it.

You can still laser-lock it:

shot_20230823_183955_1

**Comment by xezon** [comment]
Behavior looks perfect. Well done Sir.

**Comment by commy2** [comment]
Wait until you see my next trick.

File: Patch104pZH/GameFilesEdited/Data/INI/Default/Object.ini [hunk 1]
```diff
@@ -118,6 +118,13 @@ Object DefaultThingTemplate
     End
   End
 
+  ; Patch104p @fix commy2 23/08/2023 We always need this.
```
Comment (xezon) [inline Patch104pZH/GameFilesEdited/Data/INI/Default/Object.ini]: Can you explain in comment why?

`(#2277)`

File: Patch104pZH/GameFilesEdited/Data/INI/Object/AmericaVehicle.ini [hunk 2]
```diff
@@ -1303,6 +1303,11 @@ Object AmericaVehicleSpyDrone
   RadarPriority = UNIT
   KindOf = PRELOAD CAN_CAST_REFLECTIONS VEHICLE SCORE DRONE SELECTABLE INERT NO_SELECT IGNORES_SELECT_ALL ; Patch104p @bugfix commy2 03/09/2021 No longer select Spy Drone with Q-shortcut.
 
+  Behavior = StatusBitsUpgrade ModuleTag_MakeUnselectable
```
Comment (xezon) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/AmericaVehicle.ini:1307]: Comment appears to be missing for this.

File: Patch104pZH/GameFilesEdited/Data/INI/ObjectCreationList.ini [hunk 3]
```diff
@@ -10210,6 +10210,13 @@ ObjectCreationList OCL_AmericaVehicleDozer
     Disposition = ON_GROUND_ALIGNED
     Offset = X:0 Y:0 Z:0
   End
+
+  ; !!!!! debug remove
+  CreateObject
```
Comment (xezon) [inline Patch104pZH/GameFilesEdited/Data/INI/ObjectCreationList.ini]: Debug needs to be removed before merge.

**Comment by commy2** [comment]
Done. To get this to work reliably, we always need the dummy upgrade. I think the best is to grant it via the default object. That way we avoid issues where the dummy upgrade is not granted by capturing the building as GLA (dummy upgrade mainly comes from Reactors and Supply Drop Zones in main branch) or on certain custom maps.

**Comment by xezon** [comment]
I would have made these 2 different changes, so that they are also distinct in main branch.

**Comment by commy2** [comment]
Done. This change requires: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2280. Otherwise make sure to fully construct at least one Reactor (both player and opponent).

**Comment by xezon** [comment]
AmericaVehicleSpyDrone needs to grant Upgrade_DummyUpgrade if possible.

**Comment by commy2** [comment]
That would never be called. Spy Drone is spawned, not produced or constructed.

I am atm rewriting https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2280 so Command Centers grant the DummyUpgrade, which guarantees it exists in skirmish / multiplayer / the GC from the start, and before the player can spawn a Spy Drone on any of the campaign missions.

**Comment by commy2** [comment]
Finally done. Doc updated. Applied fix to the unused sub-faction Drones as well. 

Requires: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2280

File: Patch104pZH/GameFilesEdited/Data/INI/Object/LaserGeneral.ini [hunk 4]
```diff
@@ -5743,7 +5743,13 @@ Object Lazr_AmericaVehicleSpyDrone
 
   ; *** ENGINEERING Parameters ***
   RadarPriority = UNIT
-  KindOf = PRELOAD CAN_CAST_REFLECTIONS VEHICLE SCORE DRONE SELECTABLE INERT NO_SELECT IGNORES_SELECT_ALL ; Patch104p @bugfix commy2 03/09/2021 No longer select Spy Drone with Q-shortcut.
+  KindOf = PRELOAD CAN_CAST_REFLECTIONS VEHICLE SCORE DRONE SELECTABLE INERT NO_SELECT IGNORES_SELECT_ALL ; Patch104p @bugfix commy2 03/09/2021 No longer select Spy Drone with Q-shortcut. (#173)
+
+  ; Patch104p @fix commy2 23/08/2023 This makes the plane not attacked automatically and entirely unselectable. (#2277)
```
Comment (xezon) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/LaserGeneral.ini]: `drone` or `object`

File: Patch104pZH/Design/Changes/v1.0/173_spydrone_select_all_key.yaml [hunk 5]
```diff
@@ -1,10 +1,10 @@
 ---
 date: 2021-09-03
 
-title: Removes Select All key bindings from USA Spy Drone
+title: USA Spy Drone can no longer be selected
 
 changes:
-  - fix: The USA Spy Drone can no longer be selected by using the Select Aircraft (W) and Select All (Q) keys.
+  - fix: The USA Spy Drone can no longer be selected. It can still be targeted by attacks and abilities.
```
Comment (xezon) [inline Patch104pZH/Design/Changes/v1.0/173_spydrone_select_all_key.yaml]: I think this text can remain specific by saying that mouse, W and Q can no longer select it.

**Review by xezon — APPROVED** [review]

**Pull request merged**